cli/azd/pkg/azsdk/zip_deploy_client.go

```diff
@@ -113,8 +117,87 @@ func (c *ZipDeployClient) BeginDeploy(
 	return runtime.NewPoller(response, c.pipeline, pollerOptions)
 }
 
+// Deploys the specified application zip to the azure app service using deployment status api and waits for completion
+func (c *ZipDeployClient) BeginDeployTrackStatus(
+	ctx context.Context,
+	zipFile io.Reader,
+	subscriptionId,
+	resourceGroup,
+	appName string,
+	printStatus func(string),
+) (*runtime.Poller[armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse], error) {
+	printStatus("Creating deployment request")
```

Comment:
Let's remove `printStatus` for now since these may be internal details that the user might or might not be familiar with, it may be more useful to log details about the lifecycle of the application (Upload, Build, Run).